include/eosio/vm/backend.hpp

```diff
@@ -163,23 +165,32 @@ namespace eosio { namespace vm {
             }
 
             // Second pass: uses actual required memory for final parsing
-            mod.allocator.use_fixed_memory(largest_size);
-            return parser_t{ mod.allocator, options }.parse_module2(orig_ptr, sz, mod, debug);
+            mod->allocator.use_fixed_memory(largest_size);
+            return parser_t{ mod->allocator, options }.parse_module2(orig_ptr, sz, *mod, debug);
          }
       }
 
+      void share(const backend& from) {
+         mod                    = from.mod;
+         exec_ctx_created_by_backend  = from.exec_ctx_created_by_backend;
```

Comment:
This doesn't seem right. Nothing is changing `ctx` here, so `exec_ctx_created_by_backend` shouldn't change either.

Reply:
Thank you Steve @swatanabe for reviewing this! 

The "empty" backend's `exec_ctx_created_by_backend` is constructed to have the default value `true` (to support legacy uses). It needs to be set to the actual value from the instantiated module when used the first time.

Reply:
This means that `share` can only be used in a very specific way. Any other combination will result in leaks or double deletes.  This is really dangerous to begin with and it's not even documented or checked.

Reply:
Thanks. I added checks to prevent invalid sharing and added comments about the method.